pkg/services/ngalert/notifier/channels/discord.go

```diff
@@ -196,65 +197,52 @@ func (d DiscordNotifier) SendResolved() bool {
 
 func (d DiscordNotifier) constructAttachments(ctx context.Context, as []*types.Alert, embedQuota int) []discordAttachment {
 	attachments := make([]discordAttachment, 0)
-	for i := range as {
```

Comment:
Sorry that this diff is so hard to read. The changemoves all the looping code inside `forEachStoredImage`, where the database read, etc happens.